oteapi/strategies/resource/postgres.py

```diff
@@ -0,0 +1,176 @@
+    [`PostgresResourceStrategy`][oteapi.strategies.resource.application_vnd_postgres.PostgresParseStrategy]."""
+
+    user: Optional[str] = Field(None, description="postgres server username")
+    dbname: Optional[str] = Field(None, description="postgres dbname name")
+    password: Optional[str] = Field(None, description="postgres password")
+
+    sqlquery: Optional[str] = Field("", description="A SQL query string.")
+
+
+class PostgresResourceConfig(ResourceConfig):
+    """Postgresql parse strategy config"""
+
+    configuration: PostgresConfig = Field(
+        PostgresConfig(),
+        description=(
+            "Configuration for resource. " "Values in the accessURL take precedence."
+        ),
+    )
+    datacache_config: Optional[DataCacheConfig] = Field(
+        None,
+        description="Configuration options for the local data cache.",
+    )
+
+    @classmethod
+    def _urlconstruct(
+        cls,  # PEP8 - Always use cls for the first argument to class methods.
+        scheme: Optional[str] = "",  # Schema defining link format
+        user: Optional[str] = None,  # Username
+        password: Optional[str] = None,  # Password
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        path: Optional[str] = "",
+        params: Optional[str] = "",
+        query: Optional[str] = "",
+        fragment: Optional[str] = "",
+    ):
+
+        """Construct a pydantic AnyUrl based on the given URL properties"""
+
+        # Hostname should always be given
+        if not host:
+            raise Exception("hostname must be specified")
+
+        # Update netloc of username or username|password pair is defined
+        netloc = host
+        if user and not password:  # Only username is provided. OK
+            netloc = f"{user}@{host}"
+        elif user and password:  # Username and password is provided. OK
+            netloc = f"{user}:{password}@{host}"
+        else:  # Password and no username is provided. ERROR
+            raise Exception("username not provided")
+
+        # Append port if port is defined
+        netloc = netloc if not port else f"{netloc}:{port}"
+
+        # Construct a URL from a tuple of URL-properties
+        unparsed = urlunparse([scheme, netloc, path, params, query, fragment])
+
+        # Populate and return a Pydantic URL
+        return parse_obj_as(AnyUrl, unparsed)
+
+    @root_validator
+    def adjust_url(cls, values):
+        """Root Validator
+        Verifies configuration consistency, merge configurations
+        and update the accessUrl property.
+        """
+
+        # Copy model-state into placeholders
+        config = values.get("configuration")
+        accessUrl = values["accessUrl"]
+
+        # Check and merge user configuration
+        user = accessUrl.user if accessUrl.user else config["user"]
+        if config["user"] and user != config["user"]:
+            raise ValueError("mismatching username in accessUrl and configuration")
```

Comment:
Here ValueError is used, so perhaps use this exception for the previous comments on specific Exception subclass

Reply:
Done